contracts/RMRK/utils/RMRKEquipRenderUtils.sol

```diff
@@ -435,4 +423,50 @@ contract RMRKEquipRenderUtils {
             }
         }
     }
+
+    function splitSlotAndFixedParts(
+        uint64[] memory allPartIds,
+        address baseAddress
+    )
+        public
+        view
+        returns (uint64[] memory slotPartIds, uint64[] memory fixedPartIds)
+    {
+        IRMRKBaseStorage.Part[] memory allParts = IRMRKBaseStorage(baseAddress)
+            .getParts(allPartIds);
+        uint256 numFixedParts;
+        uint256 numSlotParts;
+
+        uint256 numParts = allPartIds.length;
+        // This for loop is just to discover the right size of the split arrays, since we can't create them dynamically
+        for (uint256 i; i < numParts; ) {
+            if (allParts[i].itemType == IRMRKBaseStorage.ItemType.Fixed)
+                numFixedParts += 1;
+                // We could just take the numParts - numFixedParts, but it doesn't hurt to double check it's not an uninitialized part:
+            else if (allParts[i].itemType == IRMRKBaseStorage.ItemType.Slot)
+                numSlotParts += 1;
+            unchecked {
+                ++i;
+            }
+        }
+
+        slotPartIds = new uint64[](numSlotParts);
+        fixedPartIds = new uint64[](numFixedParts);
+        uint256 slotPartsIndex;
+        uint256 fixedPartsIndex;
+
+        // This for loop is to actually fill the split arrays
```

Comment:
I wish this could have been done in the previous loop, but I get that this is impossible 😬